vector-im/element-android
# [Location sharing] - Show message on start of a live

## Type of change


## Content


Handling beacon state event in the timeline to indicate a live location share has been started. This state of the message will be displayed until a location is received. Adding support for preview of live location sharing message as well.

## Motivation and context


Closes #5710 

## Screenshots / GIFs





### Default layout mode
<img src="https://user-images.githubusercontent.com/46314705/162434013-5807c824-9ebd-4d54-9c98-e9b36e68370c.png" width=300 /><img src="https://user-images.githubusercontent.com/46314705/162434027-68e2ecaa-536a-4d63-abee-f7b85897a23d.png" width=300 />

### Bubble layout mode
<img src="https://user-images.githubusercontent.com/46314705/162767554-754e9fb1-a57f-4107-a1a7-c3d1064daca2.png" width=300 /><img src="https://user-images.githubusercontent.com/46314705/162767578-60da3203-a0c5-40c1-85e6-14f1e5a5d812.png" width=300 />

### UX
<img src="https://user-images.githubusercontent.com/46314705/162767651-f517501a-3e9d-4ebd-bdd8-aee9d10878ac.gif" width=300 />

## Tests



- Ensure the location sharing is enabled in Settings -> Preferences
- Go to a room
- Press the add button on the new message view
- Select the location icon
- Select live location sharing option
- Select a duration for the live
- Validate
- Check a message is displayed in the timeline to indicate the live has started

## Tested devices

- OS version(s): Android 11

## Checklist
